### ISSUE TITLE
difference in train/test scores

### ISSUE THREAD

**Issue by haricharan123** [Microsoft/LightGBM#646]
Please search your question on previous issues, stackoverflow or other search engines before you open a new one.

For bugs and unexpected issues, please provide following information, so that we could reproduce on our system.

## Environment info
Operating System: Ubuntu
CPU: 24 cores / 128GB Ram
C++/Python/R version: Python

## Problem Description

I seem to get good accuracy (AUC) with the training using a evaluation set.

However on prediction, I get a totally different set of results, Would be really helpful, if someone could point to as to as what i'm doing wrong. ?

## Steps to reproduce

```
params = {
    'boosting_type': 'gbdt',
    'objective': 'binary',
    'metric': ['auc','binary_error'],
    'num_leaves': 2048,  
    'learning_rate': 0.0202048,
    'feature_fraction': 0.801,
    'bagging_fraction': 0.6815,
    'scale_pos_weight':0.4,
    'min_gain_to_split':0.01
}
```

```
gbm = lgb.train(params,
                lgb_train,
                num_boost_round=10000,
                valid_sets=lgb_eval,
                verbose_eval =10,
                early_stopping_rounds=500)

```

Early stopping, best iteration is:
[392]	valid_0's binary_error: 0.199585	valid_0's auc: 0.869714

```
y_pred = gbm.predict(X_test, num_iteration=gbm.best_iteration)

for i in range(0,np.take(y_pred.shape,0)):
    if y_pred[i]>=.5:
        y_pred[i]=1
    else:  
        y_pred[i]=0

print('The accuracy of prediction is:', accuracy_score(y_test, y_pred))
print('The roc_auc_score of prediction is:', roc_auc_score(y_test, y_pred))
print('The Null Acccuracy is:', max(y_test.mean(), 1 - y_test.mean()))
```

The accuracy of prediction is: 0.740415086684
The roc_auc_score of prediction is: 0.770591966672
The Null Acccuracy is: 0.702214315809

**Comment by guolinke** [comment]
I think the AUC score should be calculated by the probability, not the 0/1 .
